app/src/main/java/com/internship/tvseries/ui/Popular/PopularViewModel.java

```diff
@@ -9,20 +9,37 @@
 import com.internship.tvseries.utils.Constants;
 
 import java.util.List;
+import java.util.function.Consumer;
 
 public class PopularViewModel extends BaseViewModel {
     private final TvRepository tvRepository;
+    private final TvRepository backendPopularRepository;
 
     private final MutableLiveData<List<Result>> _popularTvs = new MutableLiveData<>();
     public LiveData<List<Result>> popularTvs = _popularTvs;
 
-    public PopularViewModel(TvRepository tvRepository) {
+    public PopularViewModel(TvRepository tvRepository, TvRepository backendPopularRepository) {
         this.tvRepository = tvRepository;
+        this.backendPopularRepository= backendPopularRepository;
         getPopular();
     }
 
     private void getPopular() {
-        tvRepository.getByCategory(Constants.CATEGORY_POPULAR, _popularTvs::postValue);
+        tvRepository.getByCategory(Constants.CATEGORY_POPULAR, new Consumer<List<Result>>() {
```

Comment:
First time we call the backend and then the tv repo. You forgot to call the backend here. Please check for Top Rated too